internal/api/shared/database/model/dao.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
```

Comment:
this package is because you would have a circular dep ? 

Anyway I would rename it ``core`` maybe or something else as ``model`` seems to not match such interfaces

Reply:
not sure if `core` is really accurate as well, since it just contains interface and struct for the errors. It doesn't hold the actual implementation about the database.